akka-docs/src/main/paradox/scala/futures.md

```diff
@@ -243,10 +373,116 @@ it will behave as if we hadn't used the `recover` method.
 You can also use the `recoverWith` method, which has the same relationship to `recover` as `flatMap` has to `map`,
 and is use like this:
 
-@@snip [FutureDocSpec.scala]($code$/scala/docs/future/FutureDocSpec.scala) { #try-recover }
+Scala
+:  @@snip [FutureDocSpec.scala]($code$/scala/docs/future/FutureDocSpec.scala) { #try-recover }
+
+Java
+:  @@snip [FutureDocTest.java]($code$/java/jdocs/future/FutureDocTest.java) { #try-recover }
 
 ## After
 
 `akka.pattern.after` makes it easy to complete a `Future` with a value or exception after a timeout.
 
-@@snip [FutureDocSpec.scala]($code$/scala/docs/future/FutureDocSpec.scala) { #after }
\ No newline at end of file
+Scala
+:  @@snip [FutureDocSpec.scala]($code$/scala/docs/future/FutureDocSpec.scala) { #after }
+
+Java
+:  @@snip [FutureDocTest.java]($code$/java/jdocs/future/FutureDocTest.java) { #imports7 #after }
+
+@@@ div { .group-java }
```

Comment:
From here, the below section is only for Java.
Because `@@@ div { .group-java }` cannot contain `@@@ note` inside, there are multiple  `@@@ div { .group-java }` and  `@@@ note { .group-java }` blocks.